Merge pull request #1 from muzaluisa/exercise_1_branch

muzaluisa opened the pull request:
Hi, Siim

I took into account all your comments, could you have one quick look once again? :)

muzaluisa committed:
Merge pull request #1 from muzaluisa/exercise_1_branch

Exercise 1 branch

siimphh commented:
Looks like the pull request is in the opposite direction (master -> exercise_1_branch) so no diffs are showing up :)

muzaluisa committed:
Exercise 2

muzaluisa committed:
Taking into account Siim's comments

1. Added Tree class
2. Fixed dfs_path so that it returns the path from node to the end node excluding it
3. Fixed all other small comments

muzaluisa committed:
Merge pull request #4 from muzaluisa/exercise_1_branch

Merging after two review rounds

muzaluisa committed:
Merge pull request #5 from muzaluisa/exercise2_branch

Final version of exercise 2

muzaluisa committed:
Exercise 3, first version

muzaluisa committed:
Create codeU3.py

Improved docstrings

muzaluisa committed:
Refined by the first review round

Comments are provided by @windo, @luxiya01  and @blmussati , thank you for your reviews!:)
Could you please look again modified version?

siimphh commented:
I think there is something I should review but I was confused about whether this pull request contained the changes to be reviewed. So just to confirm, should I review the code as shown by this pull request?

muzaluisa committed:
First version of exercise 5 problem

Could you please review this exercise? :) 
This time I prepared only the basic implementation without extensive tests.

muzaluisa committed:
Merge pull request #6 from muzaluisa/exercise_3_branch

Exercise 3, first version

muzaluisa committed:
Exercise 5, revision 1

muzaluisa committed:
Made a test file separately

muzaluisa committed:
Delete test_rearrange_cars.py

muzaluisa committed:
Chosen combined solution by Group 8 

First version to be refactored by everyone :)

muzaluisa committed:
Minor typo fixes

muzaluisa committed:
Took into accounts Li comments on the previous ver

https://github.com/muzaluisa/codeU/pull/9#discussion_r129368376
Particularly:
- making functions private
- movement class
- typos and unused numpy library

muzaluisa committed:
printing, logging updates

muzaluisa committed:
fixed logging into the file

and added complexity explanation

muzaluisa committed:
Updated print formatting, no assert 

Took into account some comments made yesterday night by Li and Marija

muzaluisa committed:
minor changes in test_already_proper_order

from assertRaises to assertEqual

muzaluisa committed:
test_rearrange_all_cars -> test_rearrange_sample_cars

luxiya01 committed:
Add test for empty parking lot (idea from @yarynapavlyshyn)

N.B. empty parking lot test fails right now

luxiya01 committed:
Add test for empty parking lot of length 1

Test passed

luxiya01 committed:
Remove variable self.empty_position to pass test_empty_parking_lot

muzaluisa committed:
fixed test_long_parking_lot comment typo

muzaluisa committed:
minor changes in syntax suggested by pylint ..

marijja committed:
Corrected printing

luxiya01 committed:
Modify printing class and add comments

luxiya01 committed:
Add test to check that initial_order is not modified after rearranging cars

luxiya01 committed:
Modify code to past the newly added test

self.current_car_position needs to be a copy of self.initial_car_position, they previously refered to the same object

luxiya01 committed:
Modifications according to comments from @blmussati

luxiya01 committed:
Change variable names for consistency

Based on comments from @blmussati

muzaluisa committed:
made printing to console for all cases, except long_parking_lot

muzaluisa committed:
remove unnecessary to exercise 8 file

muzaluisa committed:
remove unnecessary codeU2.py file

muzaluisa committed:
remove unnecessary codeU3.py file

muzaluisa committed:
Delete Assignment1-Gettingstarted.pdf

muzaluisa committed:
Update README.md

muzaluisa committed:
Rename RearrangeCarGroup8Movements.py to RearrangeCarGroup8.py

muzaluisa committed:
edit: from RearrangeCarGroup8 import RearrangeCars

luxiya01 committed:
Correct movement printing

muzaluisa committed:
fixed list() to [] and some minor assertEqual issue

added self.assertEqual(rc_length_one.current_car_order, [0])
to compare with the desired result instead of rc_length_one.final_car_order and 
similarly in self.assertEqual(rc_empty.current_car_order, [])

muzaluisa committed:
Merge pull request #10 from muzaluisa/group_8_solution

Chosen combined solution by Group 8 :)
All people in the group have contributed to the development of our solution and I guess it was super experience for all of us! 
The basis for the solution was my code, but it was in the end quite revised.

muzaluisa committed:
Merge pull request #8 from muzaluisa/exercise_5_branch

Exercise 5 revised
